src/Halogen/DatePicker/Component/Duration.purs

```diff
@@ -0,0 +1,105 @@
+module Halogen.Datepicker.Component.Duration where
+
+import Prelude
+
+import Data.Array (fold)
+import Data.Bifunctor (bimap)
+import Data.Either (Either(..))
+import Data.Functor.Coproduct (Coproduct, coproduct, right, left)
+import Data.Generic.Rep (class Generic)
+import Data.Generic.Rep.Show (genericShow)
+import Data.Interval (Duration, IsoDuration, mkIsoDuration, unIsoDuration)
+import Data.Maybe (Maybe(..), maybe)
+import Data.Monoid (mempty)
+import Data.Monoid.Endo (Endo(..))
+import Data.Newtype (unwrap)
+import Data.Traversable (for, sequence)
+import Halogen as H
+import Halogen.Datepicker.Component.Types (BasePickerQuery(..), PickerMessage(..), PickerQuery(..), PickerValue)
+import Halogen.Datepicker.Format.Duration as F
+import Halogen.Datepicker.Internal.Num as N
+import Halogen.Datepicker.Internal.Range (minRange)
+import Halogen.Datepicker.Internal.Utils (componentProps, transitionState', asRight, mustBeMounted, pickerProps)
+import Halogen.HTML as HH
+import Halogen.HTML.Events as HE
+
+
+data DurationQuery a = UpdateCommand F.Command (Maybe Number) a
+type State = PickerValue DurationError IsoDuration
+type QueryIn = PickerQuery Unit State
+type Query = Coproduct QueryIn DurationQuery
+type Message = PickerMessage State
+
+data DurationError = InvalidIsoDuration
+derive instance durationErrorEq ∷ Eq DurationError
+derive instance durationErrorOrd ∷ Ord DurationError
+derive instance durationErrorGeneric ∷ Generic DurationError _
+instance durationErrorShow ∷ Show DurationError where
+  show = genericShow
+
+type Slot = F.Command
+
+
```

Comment:
Oh! I mean that during reading the code blank lines usually mean that things below are slightly different with previous. That's why I asked :)